lib/ansible/module_utils/basic.py

```diff
@@ -1591,6 +1591,13 @@ def _check_argument_values(self, spec=None, param=None):
             choices = v.get('choices', None)
             if choices is None:
                 continue
+            if BOOLEANS.intersection(choices):
+                msg = (
+                    'Using boolean keywords ({0}) in string parameters is discouraged. '
+                    'Update the code and API for this module. AnsibleModule will '
+                    'refuse to accept such choices in the '
+                    'future').format(','.join(map(str, BOOLEANS)))
+                self.deprecate(msg, version='2.12')
```

Comment:
This is a pretty short deprecation cycle, I thought we were supposed to wait for 2.14 at this point